src/main/java/gregtech/common/covers/CoverRoboticArm.java

```diff
@@ -180,44 +188,50 @@ private boolean shouldDisplayAmountSlider() {
+                        .overlay(GTGuiTextures.TRANSFER_MODE_OVERLAY)
+                        .build())
+                .child(new Row().right(0).coverChildrenHeight()
+                        .child(new TextFieldWidget().widthRel(0.5f).right(0)
+                                .setEnabledIf(w -> shouldDisplayAmountSlider())
+                                .setNumbers(0, Integer.MAX_VALUE)
+                                .value(filterTransferSize)
+                                .setTextColor(Color.WHITE.darker(1))));
+    }
+
+    @Override
+    protected int getMaxStackSize() {
+        return getTransferMode().maxStackSize;
+    }
+
+    @Override
+    public void writeInitialSyncData(@NotNull PacketBuffer packetBuffer) {
+        super.writeInitialSyncData(packetBuffer);
+        packetBuffer.writeInt(this.transferMode.ordinal());
     }
 
     @Override
-    protected ModularUI buildUI(Builder builder, EntityPlayer player) {
-        WidgetGroup primaryGroup = new WidgetGroup();
-        primaryGroup.addWidget(new CycleButtonWidget(91, 45, 75, 20,
-                TransferMode.class, this::getTransferMode, this::setTransferMode)
-                        .setTooltipHoverString("cover.robotic_arm.transfer_mode.description"));
-
-        ServerWidgetGroup stackSizeGroup = new ServerWidgetGroup(this::shouldDisplayAmountSlider);
-        stackSizeGroup.addWidget(new ImageWidget(111, 70, 35, 20, GuiTextures.DISPLAY));
-
-        stackSizeGroup.addWidget(
-                new IncrementButtonWidget(146, 70, 20, 20, 1, 8, 64, 512, itemFilterContainer::adjustTransferStackSize)
-                        .setDefaultTooltip()
-                        .setTextScale(0.7f)
-                        .setShouldClientCallback(false));
-        stackSizeGroup.addWidget(new IncrementButtonWidget(91, 70, 20, 20, -1, -8, -64, -512,
-                itemFilterContainer::adjustTransferStackSize)
-                        .setDefaultTooltip()
-                        .setTextScale(0.7f)
-                        .setShouldClientCallback(false));
-
-        stackSizeGroup.addWidget(new TextFieldWidget2(113, 77, 31, 20,
-                () -> String.valueOf(itemFilterContainer.getTransferStackSize()), val -> {
-                    if (val != null && !val.isEmpty())
-                        itemFilterContainer.setTransferStackSize(
-                                MathHelper.clamp(Integer.parseInt(val), 1, transferMode.maxStackSize));
-                })
-                        .setNumbersOnly(1, transferMode.maxStackSize)
-                        .setMaxLength(4)
-                        .setScale(0.9f));
-
-        primaryGroup.addWidget(stackSizeGroup);
-
-        return super.buildUI(builder.widget(primaryGroup), player);
+    public void readInitialSyncData(@NotNull PacketBuffer packetBuffer) {
+        super.readInitialSyncData(packetBuffer);
+        this.transferMode = TransferMode.values()[packetBuffer.readInt()];
```

Comment:
It would be preferable to have a `public static final` array of `TransferMode`'s values, similar to `EnumFacing`. `Enum#values` creates a new array each time.